Updated authorization docs (#64)
revisit-studies/reVISit-studies.github.io: docs/data-and-deployment/firebase-setup.md

![Console](./img/firebase_steps/step15.jpg)

:::info

> In addition to what is shown above, you will want to enable "Google Authentication" in the same section of Firebase. You can do this by choosing the "Google" sign in method under the "Additional Providers" section. The anonymous sign-in is used for particpants taking the study. It allows them to access and send data to the Firestore without external authentication. The Google sign-in method is used to authenticate administrators for managing studies and data. Please see [here](../authentication-authorization/adding-removing-ui) for more detailed authentication information.

Reviewer: ```suggestion
In addition to what is shown above, you will want to enable "Google Authentication" in the same section of Firebase. You can do this by choosing the "Google" sign in method under the "Additional Providers" section. The anonymous sign-in is used for particpants taking the study. It allows them to access and send data to the Firestore without external authentication. The Google sign-in method is used to authenticate administrators for managing studies and data. Please see [here](../authentication-authorization/adding-removing-ui) for more detailed authentication information and set up instructions.
```

### Allowing for CORS requests

Once that is finished, we'll have to use Google's `gsutil` function in the terminal to set up a CORS policy so that the application can communicate with Firebase storage. Follow <a href="https://cloud.google.com/storage/docs/gsutil_install" targe="_blank">these steps on how to install gsutil on your local machine</a>.

After installing gsutil, you need to navigate to the `google-cloud-sdk/bin` folder on your local machine. Create a new file called "cors.json" with the following contents:

```js
[
  {
    origin: ["*"],
    method: ["GET"],

>     maxAgeSeconds: 3600,
>   },
> ];

Reviewer: Do we need these? Will it break stuff?
```suggestion
    maxAgeSeconds: 3600
  }
]
```

```

Lastly, while still inside this same directory, call the following function:

```bash
gsutil cors set cors.json gs://<your-cloud-storage-bucket>
```

You can find the link for the storage bucket by navigating to the "storage" product in Firebase.

![Console](./img/firebase_steps/gsutil_step1.jpg)

Assuming that you have already changed the `VITE_STORAGE_ENGINE` variable in the `.env` file to "firebase", you can now call `yarn serve` to launch the server and navigate to http://localhost:8080 to launch any demo study. All data from any participation will automatically be uploaded to the store.

:::note
When running `yarn build`, reVISit automatically uses the Firebase storage engine.
:::

![Demo](./img/demo.png)

### Deployment

> If you intending to deploy your reVISit application to a static website, we suggest following our [guide to deploy to Github pages](../deploying-to-static-website). In this, you'll see that there is an additional step that you will need to follow in Firebase so that your custom domain is authorized with Firebase.

Reviewer: ```suggestion
If you intending to deploy your reVISit application to a static website, we suggest following our [guide to deploy to Github pages](../deploying-to-static-website). In this, you'll see that there is an additional step that you will need to follow in Firebase so that your custom domain can authorize users.
```
